### PR TITLE
Validator

### PR DESCRIPTION
**Pull request by bmbrina**
## What does this PR do?

* Adds some validations from validator.js

### CODE REVIEW

**Comment by walreyes** [comment]
Alright @bmbrina 

I've added the specs for the Validator.

We will follow a `Test Driven Development` approach, make those specs green and all the work is done with the Validator.

The specs live inside `test/validator` you can enter the specs and see what i'm testing.

To run the specs in the terminal run the following commands:

```shell
docker-compose run jsmodels bash
npm test
```

The specs get executed everytime you update a file.

File: validator/index.js [hunk 1]
```diff
@@ -1,29 +1,109 @@
...
-    return {
-      numeric: this.isNumeric,
-      email: validator.isEmail
+  constructor (values = {}, validations = {}, CustomValidator = null) {
+    this.values = values
+    this.validations = validations
+    this.validationsDictionary = Validator.VALIDATIONS_DICTIONARY()
+
+    if (CustomValidator) {
+      this.validationsDictionary = Object.assign({}, this.validationsDictionary, CustomValidator.VALIDATIONS_DICTIONARY())
```
Comment (walreyes) [inline validator/index.js]: Mobe `Object.assign({}, this.validationsDictionary, CustomValidator.VALIDATIONS_DICTIONARY())` to it's own function `mergeCustomValidatorDictionary`

```javascript
this.validationsDictionary = this.mergeCustomValidatorDictionary(this.validationsDictionary, CustomValidator)
```

File: validator/index.js [hunk 2]
```diff
@@ -1,29 +1,109 @@
...
-    console.log("I am numeric!")
+  static validate (values = {}, validations = {}, customValidator = null) {
+    const validator = new Validator(values, validations, customValidator)
+    return validator.runValidations()
   }
 
-  isNumeric() {
-    console.log("It is numeric")
+  runValidations () {
+    // Run all your validations in here.
```
Comment (walreyes) [inline validator/index.js]: Remove this comments, please

File: validator/index.js [hunk 3]
```diff
@@ -1,29 +1,109 @@
...
+    return { valid: validAttribute, errors }
+  }
+
+
+  runValidation (validationKey, validation, value, params) {
+    let passed = true
+    let error
+
+    // Check that validation exists
+    if (!validation) {
```
Comment (walreyes) [inline validator/index.js]: Let's move the validations to function.

For example, move: `!validation.validate || typeof(validation.validate) != 'function'` to a function called `isAFunction(variable)`

```javascript
isAFunction(variable){
  return validation.validate && typeof(validation.validate) === 'function'
}
```

And then you could call `if(!isAFunction(validation.validate)`

That way it's more clear what it's being evaluated.

You could do the same with `typeof(passed) != 'boolean'`.

File: validator/validations_dictionaries/ValidatorDictionary.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+var validator = require('validator');
+
+class ValidatorDictionary {
```
Comment (walreyes) [inline validator/validations_dictionaries/ValidatorDictionary.js]: 👌

File: validator/index.js (hunk 2 above)
Comment (bmbrina) [reply validator/index.js]: 👌🏻

File: validator/index.js (hunk 1 above)
Comment (bmbrina) [reply validator/index.js]: 👌🏻

**Comment by bmbrina** [comment]
@walreyes the changes are done, I'm only missing `the README.md`, should I make it at the root and name it `Validator.md` or add a section to the main readme?

File: validator/index.js (hunk 3 above)
Comment (walreyes) [reply validator/index.js]: You could even have used `Validator.validate()` to check if it's a function or if it's boolean.

Validator would validate itself.

I'm joking but it'd be cool

**Pull request merged**